### PR TITLE
Add settings feature

### LINKED ISSUE #89: Add settings feature to application

### PR DESCRIPTION
**Pull request by khoodehui**
`SetCommand` will modify `UserPrefs`, which in turn updates `preferences.json` upon exiting the application.

Also implemented functionality to allow the new settings to be reflected immediately, however the design of it is not quite desirable for now (see `SettingsUpdateManager.java`), might update it later on.

Light mode looks like this:
Screenshot 2020-10-28 at 3 29 58 PM

### CODE REVIEW

File: src/main/java/seedu/momentum/commons/core/StatisticTimeframe.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    /**
+     * Converts the timeframe to a {@code ChronoUnit}.
+     */
+    public ChronoUnit toChronoUnit() {
+        return timeframe == Timeframe.DAILY
+            ? ChronoUnit.DAYS
+            : timeframe == Timeframe.WEEKLY
+            ? ChronoUnit.WEEKS
+            : ChronoUnit.MONTHS;
+    }
```
Comment (claracheong4) [inline src/main/java/seedu/momentum/commons/core/StatisticTimeframe.java]: Implement the conversion from enum Timeframe to ChronoUnit in the enum class instead then call the method here?

File: src/main/java/seedu/momentum/commons/core/StatisticTimeframe.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+    @Override
+    public String toString() {
+        return timeframe == Timeframe.DAILY
+            ? "Daily"
+            : timeframe == Timeframe.WEEKLY
+            ? "Weekly"
+            : "Monthly";
+    }
```
Comment (claracheong4) [inline src/main/java/seedu/momentum/commons/core/StatisticTimeframe.java]: Override toString method in TimeFrame to something like `name().charAt(0) + name().substring(1).toLowerCase()` ?

File: src/main/java/seedu/momentum/logic/parser/SetCommandParser.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import seedu.momentum.model.Model;
+
+public class SetCommandParser implements Parser<SetCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the SetCommand
+     * and returns an SetCommand object for execution.
+     *
+     * @param model the current model.
+     * @throws ParseException if the user input does not conform the expected format
```
Comment (claracheong4) [inline src/main/java/seedu/momentum/logic/parser/SetCommandParser.java]: @throws should end with a .

**Review by claracheong4 — APPROVED** [review]
Other than some minor stuff, everything looks alright.

**Review by boundtotheearth — APPROVED** [review]
Other than clara's comments, LGTM

**Pull request merged**